### PR TITLE
[docs] Add search.md

### PR DESCRIPTION
**Pull request by ras0219-msft**
Depends on PR #24754. Adds documentation for the `vcpkg search` command.

### CODE REVIEW

**Comment by BillyONeal** [comment]
Since `search` is deprecated `find port`, should we be directing people there instead?

File: docs/commands/search.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+vcpkg search [options] [query]
+```
+
+## Description
+
+Searches for available packages by name and description.
+
+Search performs a case-insensitive search through all available package names and descriptions. The results are displayed in a tabular format.
+
+Only packages currently available based on the [Registries Configuration][Registries] will be displayed.
```
Comment (BillyONeal) [inline docs/commands/search.md:17]: Is that actually true? I was under the impression it searched only the builtin registry. (In the one liner install it fails IIRC)

File: docs/commands/search.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+All vcpkg commands support a set of [common options](common-options.md).
+
+### `--x-full-desc`
+
+**Experimental and may change or be removed at any time**
+
+Do not truncate long descriptions.
+
+By default, long descriptions will be truncated to keep the tabular output browsable.
+
```
Comment (BillyONeal) [inline docs/commands/search.md:38]: I believe this is one of the commands that listens to `--x-json` so we should have the schema for that here.

**Review by BillyONeal — APPROVED** [review]

File: docs/commands/search.md (hunk 1 above)
Comment (ras0219-msft) [reply docs/commands/search.md:17]: I tested it as part of writing these docs.

File: docs/commands/search.md (hunk 2 above)
Comment (ras0219-msft) [reply docs/commands/search.md:38]: So I tested that and it didn't work -- but it turns out it _does_ work if you don't specify a query 🤷 . I think leaving it undocumented is better for now.

File: docs/commands/search.md (hunk 1 above)
Comment (ras0219-msft) [reply docs/commands/search.md:17]: Ok, confirmed that with vcpkg-configuration.json as:
```json
{
    "default-registry": null
}
```
I get
```
@ras0219-msft ➜ /workspaces/vcpkg (dev/roschuma/docs-integrate ✗) $ ./vcpkg search
The result may be outdated. Run `git pull` to get the latest results.

If your port is not listed, please open an issue at and/or consider making a pull request:
    https://github.com/Microsoft/vcpkg/issues
```

Comment (BillyONeal) [reply docs/commands/search.md:17]: The test I would like to see is whether it can display something that is not in any built in registry.

**Review by BillyONeal — APPROVED** [review]

**Pull request merged**